### PR TITLE
[WIP]Sync addons with latest upstream, and update documentation

### LINKED ISSUE #868: DNS addon in guide is out of date
The DNS addon in the guide is completely out of date; for example it installs a ReplicationController which has been long deprecated in favour of Deployment. Check [the current version of DNS addon](https://github.com/kubernetes/kubernetes/blob/release-1.6/cluster/addons/dns/).

### PR DESCRIPTION
**Pull request by aknuds1**
**NOTE: This is not yet working**

The Kubernetes addons in the documentation are completely out of date, for example they use ReplicationController which has long been deprecated in favour of Deployment. I've synced the addons to version 1.6 of the upstream repository and updated the documentation correspondingly. Fixes #868

The installed DNS addon is sadly not working for me, and I could really use some help in figuring out why. I suspect the issue is that the API server isn't reachable on https://10.3.0.1 (see #870).

Basically, I'm seeing the following errors in the kubedns container in the kube-dns pod:

    E0419 22:06:50.952662       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: Get https://10.3.0.1:443/api/v1/services?resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
    I0419 22:07:20.444655       1 dns.go:174] DNS server not ready, retry in 500 milliseconds
    F0419 22:07:20.944622       1 dns.go:168] Timeout waiting for initialization

### CODE REVIEW

**Comment by 5tefan** [comment]
Was seeing similar "Failed to list *v1.Service type errors" in another project, fixed with hermanjunge/kubernetes-digitalocean-terraform#31 and is possibly related to #865.

**Comment by aknuds1** [comment]
@5tefan Thanks for the input! I tried making the adjustments in [your PR](https://github.com/hermanjunge/kubernetes-digitalocean-terraform/pull/31), but it didn't make any difference :( Looks as if 10.3.0.1 isn't available within the cluster.

**Comment by aknuds1** [comment]
@aaronlevy Hey, I hear you might be able to help out with this? I.e., you might understand why API Server isn't on https://10.3.0.1 after being set up?

**Comment by dpgeekzero** [comment]
FYI: I setup a baremetal cluster today following the coreos docs, but I used this for kube-dns.

It's working great, no problems here.

**Comment by aknuds1** [comment]
@dpgeekzero Thanks for testing 🙂

**Comment by Ascendance** [comment]
would love to hear how this pans out

**Comment by Ascendance** [comment]
@dpgeekzero Did you use flannel or Calico?

**Comment by zbwright** [comment]
Hey - Thanks for your help on these. I'm closing this PR because we've completely rewritten and moved the Kubernetes documentation. Take a look, if you like: https://coreos.com/tectonic/docs/latest/tutorials/kubernetes/getting-started.html
